### ISSUE TITLE
[BUG] The gradient computed for bias in tf.contrib.slim.conv2d + tf.contrib.slim.batch_norm and tf.contrib.slim.conv2d(normalizer=tf.contrib.slim.batch_norm) are different!

### ISSUE THREAD

**Issue by yxchng** [tensorflow/tensorflow#33754]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04.3 LTS
- TensorFlow version (use command below): tensorflow==1.15.0rc2
- Python version: 3.6.8
- CUDA/cuDNN version: 10.1/7.5
- GPU model and memory: RTX 2080 Ti

**Describe the current behavior**
The gradient computed for bias in tf.contrib.slim.conv2d + tf.contrib.slim.batch_norm and tf.contrib.slim.conv2d(normalizer=tf.contrib.slim.batch_norm) are different!

**Describe the expected behavior**
The gradients should be the same. Right now, it seems like `tf.contrib.slim.conv2d + tf.contrib.slim.batch_norm` gives the correct gradient.

**Code to reproduce the issue**

```
import tensorflow as tf
import tensorflow.contrib.slim as slim
import numpy as np

# tf.enable_eager_execution()

print("GRAPH 1")
graph_1 = tf.Graph()
with graph_1.as_default():
    input_placeholder_1 = tf.placeholder(tf.float32, shape=([1, 10, 10, 3]))
    target_placeholder_1 = tf.placeholder(tf.float32, shape=([1, 10, 10, 1]))

    input_tensor_1 = tf.pad(input_placeholder_1, [[0, 0], [1, 1], [1, 1], [0, 0]])
    # forward_op = slim.conv2d(inputs=input_tensor, num_outputs=1, kernel_size=3, stride=1, padding="VALID", normalizer_fn=slim.batch_norm, normalizer_params={"center": True, "scale": True}, weights_initializer=tf.constant_initializer(value=0.0), biases_initializer=None)
    forward_op_1 = slim.conv2d(inputs=input_tensor_1, num_outputs=1, kernel_size=3, stride=1, padding="VALID", normalizer_fn=None, weights_initializer=tf.constant_initializer(value=0), biases_initializer=None)
    forward_op_1 = slim.batch_norm(forward_op_1, scale=True)
    forward_op_1 = tf.sigmoid(forward_op_1)

    loss_1 = tf.losses.sigmoid_cross_entropy(target_placeholder_1, forward_op_1, reduction="weighted_sum")
    backward_op_1 = tf.gradients(loss_1, tf.trainable_variables())
    sess_1 = tf.Session(graph=graph_1)
    sess_1.run(tf.global_variables_initializer())
    train_vars_1 = sess_1.run(tf.trainable_variables())
    for train_var_1 in train_vars_1:
        print(train_var_1)

print("GRAPH 2")
graph_2 = tf.Graph()
with graph_2.as_default():
    input_placeholder_2 = tf.placeholder(tf.float32, shape=([1, 10, 10, 3]))
    target_placeholder_2 = tf.placeholder(tf.float32, shape=([1, 10, 10, 1]))

    input_tensor_2 = tf.pad(input_placeholder_2, [[0, 0], [1, 1], [1, 1], [0, 0]])
    forward_op_2 = slim.conv2d(inputs=input_tensor_2, num_outputs=1, kernel_size=3, stride=1, padding="VALID", normalizer_fn=slim.batch_norm, normalizer_params={"center": True, "scale": True}, weights_initializer=tf.constant_initializer(value=0.0), biases_initializer=None)
    # forward_op_2 = slim.conv2d(inputs=input_tensor_2, num_outputs=1, kernel_size=3, stride=1, padding="VALID", normalizer_fn=None, weights_initializer=tf.constant_initializer(value=0), biases_initializer=tf.constant_initializer(value=0))
    # forward_op_2 = slim.batch_norm(forward_op_2, scale=True)
    forward_op_2 = tf.sigmoid(forward_op_2)

    loss_2 = tf.losses.sigmoid_cross_entropy(target_placeholder_2, forward_op_2, reduction="weighted_sum")
    backward_op_2 = tf.gradients(loss_2, tf.trainable_variables())
    sess_2 = tf.Session(graph=graph_2)
    sess_2.run(tf.global_variables_initializer())
    train_vars_2 = sess_2.run(tf.trainable_variables())
    for train_var_2 in train_vars_2:
        print(train_var_2)

for i in range(10):
    print("RUN ", i)
    input_ = np.random.randn(1, 10, 10, 3)
    target = np.random.randint(2, size=(1, 10, 10, 1))
    grad_1 = sess_1.run(backward_op_1, feed_dict={input_placeholder_1: input_, target_placeholder_1: target})
    print(len(grad_1))
    print(grad_1)
    grad_2 = sess_2.run(backward_op_2, feed_dict={input_placeholder_2: input_, target_placeholder_2: target})
    print(len(grad_2))
    print(grad_2)
```
**logs**
```
GRAPH 1
[[[[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]]


 [[[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]]


 [[[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]]]
[1.]
[0.]
GRAPH 2
[[[[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]]


 [[[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]]


 [[[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]

  [[0.]
   [0.]
   [0.]]]]
[1.]
[0.]
RUN  0
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([3.8114848], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  1
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([1.3114842], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  2
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([1.3114835], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  3
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([4.811485], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  4
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.8114846], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  5
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([4.3114843], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  6
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([1.8114848], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  7
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.06148279], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  8
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([3.0614848], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
RUN  9
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([4.0614843], dtype=float32)]]
1
[[array([[[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]],


       [[[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]],

        [[0.],
         [0.],
         [0.]]]], dtype=float32), array([0.], dtype=float32), array([0.], dtype=float32)]]
```

**Other info **
I understand that tf.contrib.slim is no longer being maintained but I just want to know if it is really a bug or a misuse. Right now, based on my minimal testing code, it seems like a bug.
